googlesource/googlesource.go

```diff
@@ -128,40 +134,50 @@ func (s *Source) ReadGoogleRow(rowInput chan readRowInput, responseCh chan sdk.R
 }
 
 func (s *Source) wrtieLatestPosition(postion Position) {
-	s.LatestPositions.lock.Lock()
-	s.LatestPositions.LatestPositions[postion.TableID] = postion
-	s.LatestPositions.lock.Unlock()
+	s.latestPositions.lock.Lock()
+	s.latestPositions.LatestPositions[postion.TableID] = postion
+	s.latestPositions.lock.Unlock()
 }
 
-// runGetRow sync data for bigquery using bigquery client jobs
-func (s *Source) runGetRow(offset int, tableID string) (it *bigquery.RowIterator, err error) {
-	q := s.BQReadClient.Query(
-		"SELECT * FROM `" + s.SourceConfig.Config.ConfigProjectID + "." + s.SourceConfig.Config.ConfigDatasetID + "." + tableID + "` " +
-			"LIMIT " + strconv.Itoa(googlebigquery.CounterLimit) + " OFFSET " + strconv.Itoa(offset))
+// getRowIterator sync data for bigquery using bigquery client jobs
+// haris proposal to rename to getRowIterator, since it's not returning a single row
+// Neha: DONE
+func (s *Source) getRowIterator(offset int, tableID string) (it *bigquery.RowIterator, err error) {
+	// haris: does BigQuery guarantee ordering?
+	// Neha: DONE. it does not guarantee ordering and so have added a config where user can provide the column name which
+	// would be used as orderBy value. Orderby is not mandatory though
```

Comment:
How about insertion ordering?

Reply:
Will be handled using `incrementing.column`